### PR TITLE
Added missing fields related to customer information

### PR DESCRIPTION
**Pull request by yogeshsuhagiya**
## This PR is a:

- [ ] New topic
- [x] Content update
- [ ] Content fix or rewrite
- [x] Bug fix or improvement

## Summary

- Added missing fields related to customer information.
- Format the table structure code,

## Additional information

List all affected URLs 

- https://devdocs.magento.com/guides/v2.3/architecture/gdpr/magento-2x.html
- https://devdocs.magento.com/guides/v2.3/architecture/archi_perspectives/framework.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/architecture/gdpr/magento-2x.md [hunk 1]
```diff
@@ -229,16 +232,21 @@ The following columns in the `quote` table contain customer information:
...
 --- | ---
-`customer_id` | int(10)
-`customer_email` | varchar(255)
-`customer_prefix` | varchar(40)
-`customer_firstname` | varchar(255)
-`customer_middlename` | varchar(40)
-`customer_lastname` | varchar(255)
 `customer_dob` | datetime
+`customer_email` | varchar(128)
+`customer_firstname` | varchar(128)
```
Comment (rogyar) [inline guides/v2.1/architecture/gdpr/magento-2x.md]: Please, make sure that the field sizes are correct here and everywhere. Take a look at 
https://github.com/magento/magento2/blob/1f9186c3b9a96c5e642fd5d3d31ac5c7e1877d2b/app/code/Magento/Quote/Setup/InstallSchema.php#L179
Wher the customer_firstname is VARCHAR(255)

File: guides/v2.1/architecture/archi_perspectives/framework.md [hunk 2]
```diff
@@ -102,72 +103,71 @@ Other namespaces under `Magento\Framework` that will interest extension develope
...
+            <td>Contains the basic exceptions that are thrown throughout the Magento codebase.</td>
+        </tr>
+        <tr>
+            <td>
+                <code>Magento\Framework\Event</code>
+            </td>
+            <td>Contains the code that publishes synchronous events and that handles observers for any Magento event is handled here.</td>
+        </tr>
 		<tr>
 			<td>
```
[the remark this answers is not in the document]
Comment (yogeshsuhagiya) [reply guides/v2.1/architecture/archi_perspectives/framework.md]: Hi @rogyar Adjusted the content here too.

File: guides/v2.1/architecture/gdpr/magento-2x.md (hunk 1 above)
Comment (yogeshsuhagiya) [reply guides/v2.1/architecture/gdpr/magento-2x.md]: @rogyar Corrected data types of fields like customer_firstname, customer_lastname, etc..

**Comment by yogeshsuhagiya** [comment]
Hi @rogyar I've made changes as per your request, can you please review it?

**Comment by dobooth** [comment]
Need architect review since it is GDPR.

**Comment by keharper** [comment]
This topic contains GDPR compliance information. It is not meant to state the entire contents of certain database tables.

File: guides/v2.1/architecture/gdpr/magento-2x.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ Magento 2 stores these customer attributes:
 
 #### `customer_entity` and reference tables
```
Comment (shrielenee) [inline guides/v2.1/architecture/gdpr/magento-2x.md]: Remove changes to this file from the PR. This is GDPR compliance and we need to control it.

**Comment by keharper** [comment]
I went through what notes I still have from a year ago. Any column omissions were deliberate, based on their role in maintaining GDPR compliance. @yogeshsuhagiya  you can keep the edits where you added a space before the word "table", but all technical changes in the `magento-2x.md` file must be backed out.

**Comment by yogeshsuhagiya** [comment]
@keharper okay I'll revoke the changes.

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=4164) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=4164) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: yogeshsuhagiya<br/>:x: yogeshaureatelabs

File: guides/v2.1/architecture/gdpr/magento-2x.md (hunk 3 above)
Comment (yogeshsuhagiya) [reply guides/v2.1/architecture/gdpr/magento-2x.md]: Removed changes

**Comment by keharper** [comment]
@yogeshsuhagiya  In general, we're trying to remove HTML tables from our files. If you work on simple tables in the future, it'd be better to convert them to markdown, rather than adjusting the spacing/layout of HTML tags. I won't ask you to do that here, just something to keep in mind.

**Pull request merged**